code/modules/xenomorph/xeno_structures.dm

```diff
@@ -1175,3 +1175,238 @@ TUNNEL
+	. = ..()
+	if(!.)
+		return FALSE
+	if(!mature && isxeno(user))
+		to_chat(user, span_xenowarning("[src] hasn't grown yet, give it some time!"))
+		return FALSE
+	return TRUE
+
+/obj/structure/xeno/plant/update_icon_state()
+	. = ..()
+	icon_state = mature_icon_state
+
+///Called whenever someone uses the plant, xeno or marine
+/obj/structure/xeno/plant/proc/on_use(mob/user)
+	return TRUE
+
+///Called when the plant reaches maturity
+/obj/structure/xeno/plant/proc/on_mature(mob/user)
+	playsound(src, "alien_resin_build", 25)
+	mature = TRUE
+	update_icon_state()
+
+/obj/structure/xeno/plant/attack_hand(mob/living/user)
+	if(!can_interact(user))
+		return ..()
+	return on_use(user)
+
+/obj/structure/xeno/plant/attack_alien(mob/living/carbon/xenomorph/X, damage_amount = X.xeno_caste.melee_damage, damage_type = BRUTE, damage_flag = "", effects = TRUE, armor_penetration = 0, isrightclick = FALSE)
+	if((X.status_flags & INCORPOREAL))
+		return FALSE
+
+	if(X.a_intent == INTENT_HARM && isxenodrone(X))
+		to_chat(X, span_xenowarning("We uproot [src]!"))
+		deconstruct(FALSE)
+		return FALSE
+	if(can_interact(X))
+		return on_use(X)
+	return TRUE
+
+/obj/structure/xeno/plant/heal_fruit
+	name = "life fruit"
+	desc = "It would almost be appetizing wasn't it for the green colour and the shifting fluids inside..."
+	icon_state = "heal_fruit_immature"
+	mature_icon_state = "heal_fruit"
+	///Minimum amount of health recovered
+	var/healing_amount_min = 125
+	///Maximum amount of health recovered, depends on the xeno's max health
+	var/healing_amount_max_health_scaling = 0.5
+
+/obj/structure/xeno/plant/heal_fruit/on_use(mob/user)
+	to_chat(user, span_warning("We begin consuming [src]..."))
+	if(!do_after(user, 2 SECONDS, FALSE, src))
+		return FALSE
+	if(!isxeno(user))
+		var/datum/effect_system/smoke_spread/xeno/acid/plant_explosion = new(get_turf(src))
+		plant_explosion.set_up(3,src)
+		plant_explosion.start()
+		visible_message(span_danger("[src] bursts, releasing toxic gas!"))
+		qdel(src)
+		return TRUE
+
+	var/mob/living/carbon/xenomorph/X = user
+	var/heal_amount = max(healing_amount_min, healing_amount_max_health_scaling * X.xeno_caste.max_health)
+	HEAL_XENO_DAMAGE(X,heal_amount)
+	to_chat(X, span_xenowarning("We feel a sudden soothing chill as [src] tends to our wounds."))
+	qdel(src)
+	return TRUE
+
+/obj/structure/xeno/plant/armor_fruit
+	name = "hard fruit"
+	desc = "The contents of this fruit are protected by a tough outer shell."
+	icon_state = "armor_fruit_immature"
+	mature_icon_state = "armor_fruit"
+	///How much total sunder should we remove
+	var/sunder_removal = 30
+
+/obj/structure/xeno/plant/armor_fruit/on_use(mob/user)
+	to_chat(user, span_warning("We begin consuming [src]..."))
+	if(!do_after(user, 2 SECONDS, FALSE, src))
+		return FALSE
+	if(!isxeno(user))
+		var/turf/far_away_lands = get_turf(user)
+		for(var/x in 1 to 20)
+			far_away_lands = get_step(far_away_lands, REVERSE_DIR(user.dir))
```

Comment:
You can probably use locate(x, y, z).

You need to check if locate find a turf though, and use that code if it doesn't